Mathlib/Data/Set/Semiring.lean

```diff
@@ -166,11 +242,24 @@ instance [CommMonoid α] : CanonicallyOrderedCommSemiring (SetSemiring α) :=
 with respect to the pointwise operations on sets. -/
 def imageHom [MulOneClass α] [MulOneClass β] (f : α →* β) : SetSemiring α →+* SetSemiring β
     where
-  toFun := image f
+  toFun s := up (image f (down s))
   map_zero' := image_empty _
-  map_one' := by rw [image_one, map_one, singleton_one]
+  map_one' := by
+    dsimp only  -- porting note: structures do not do this automatically any more
+    rw [down_one, image_one, map_one, singleton_one, up_one]
   map_add' := image_union _
   map_mul' _ _ := image_mul f
 #align set_semiring.image_hom SetSemiring.imageHom
 
+-- TODO; copy formatting from mathport
+@[simp]
+lemma down_imageHom [MulOneClass α] [MulOneClass β] (f : α →* β) (s : SetSemiring α) :
+  down (imageHom f s) = f '' down s := rfl
+#align set_semiring.down_image_hom SetSemiring.down_imageHom
+
+@[simp]
+lemma imageHom_up [MulOneClass α] [MulOneClass β] (f : α →* β) (s : Set α) :
+  imageHom f (up s) = up (f '' s) := rfl
+#align set_semiring.image_hom_up SetSemiring.imageHom_up
```

Comment:
This has since been adjusted in the mathlib3 PR; I will update it here once that goes through.